rcl/src/rcl/lexer.c

```diff
@@ -650,10 +650,11 @@ rcl_lexer_analyze(
       movement = state->else_movement;
     }
 
-    // Move the lexer to another character in the string
     if (0u == movement) {
-      // Go forwards 1 char
-      ++(*length);
+      if ('\0' != current_char) {
```

Comment:
is it guaranteed that if `current_char == '\0'` the loop will end in this iteration?
I'm wondering if doing a `++(*length)` was helping the loop to be break in that case or not.

Reply:
> is it guaranteed that if current_char == '\0' the loop will end in this iteration?

I think so as long as the state machine doesn't have a state with a transition from `\0` to a non-terminal state. In the case this bug is fixing it reaches the terminal state (`T_NONE`).